src/main/java/com/google/api/gax/grpc/BundlingSettings.java

```diff
@@ -193,24 +155,29 @@ public Builder setRequestByteLimit(Integer requestByteLimit) {
      * a number higher than the number of threads that are capable of blocking on the bundler, or
      * else your application will suffer dead time while it waits for the delay threshold to trip.
      */
-    public abstract Builder setBlockingCallCountThreshold(Long blockingCallCountThreshold);
+    public abstract Builder setBlockingCallCountThreshold(long blockingCallCountThreshold);
 
     @Deprecated
     public Builder setBlockingCallCountThreshold(Integer blockingCallCountThreshold) {
       return setBlockingCallCountThreshold(blockingCallCountThreshold.longValue());
     }
 
-    /**
-     * Set if the bundling should be enabled. If set to false, the bundling logic will be disabled
-     * and the simple API call will be used. Default to true.
-     */
-    public abstract Builder setIsEnabled(Boolean enabled);
-
     abstract BundlingSettings autoBuild();
 
     /** Build the BundlingSettings object. */
     public BundlingSettings build() {
-      return autoBuild();
+      BundlingSettings settings = autoBuild();
+      Preconditions.checkArgument(
+          settings.getElementCountThreshold() > 0, "elementCountThreshold must be positive");
+      Preconditions.checkArgument(
+          settings.getRequestByteThreshold() > 0, "requestByteThreshold must be positive");
+      Preconditions.checkArgument(
+          settings.getDelayThreshold().compareTo(Duration.ZERO) > 0,
+          "delayThreshold must be positive");
+      Preconditions.checkArgument(
+          settings.getBlockingCallCountThreshold() > 0,
+          "blockingCallCountThreshold must be positive");
+      return settings;
```

Comment:
So I know that we agreed to require all of these settings in https://github.com/googleapis/gax-java/issues/168 , but I think I want to retract that. I don't know that it makes sense to require all of these for all use cases (since this will be used outside of PubSub too).